qiskit_cold_atom/fermions/ffsim_backend.py

```diff
@@ -0,0 +1,388 @@
+from qiskit_aer import AerJob
+from qiskit_nature.operators.second_quantization import FermionicOp
+from scipy.sparse.linalg import expm_multiply
+
+from qiskit_cold_atom.circuit_tools import CircuitTools
+from qiskit_cold_atom.fermions.base_fermion_backend import BaseFermionBackend
+from qiskit_cold_atom.fermions.fermion_gate_library import (
+    FermionicGate,
+    Hop,
+    Interaction,
+    LoadFermions,
+    Phase,
+)
+
+
+class FfsimBackend(BaseFermionBackend):
+    """Fermionic simulator backend that uses ffsim.
+
+    This is a specialized simulator backend for fermionic circuits that uses `ffsim`_.
+    It computes the statevector and simulate measurements with vastly improved efficiency
+    compared with the :class:`~.FermionSimulator` backend. However, unlike
+    :class:`~.FermionSimulator`,
+    it does not compute the full unitary of a circuit.
+
+    This backend is not supported on Windows, and a special ``pip`` command is needed
+    to install it:
+
+    .. code::
+
+        pip install "qiskit-cold-atom[ffsim]"
+
+    .. _ffsim: https://github.com/qiskit-community/ffsim
+    """
+
+    _DEFAULT_CONFIGURATION = {
+        "backend_name": "ffsim_simulator",
+        "backend_version": "0.0.1",
+        "n_qubits": 100,
+        "basis_gates": None,
+        "gates": [],
+        "local": False,
+        "simulator": True,
+        "conditional": False,
+        "open_pulse": False,
+        "memory": True,
+        "max_shots": 1e6,
+        "coupling_map": None,
+        "description": "ffsim simulator for fermionic circuits. Instead of qubits, each wire represents"
+        " a single fermionic mode",
+        "supported_instructions": None,
+    }
+
+    def __init__(self, config_dict: Dict[str, Any] = None, provider=None):
+        """Initializing the backend from a configuration dictionary"""
+
+        if config_dict is None:
+            config_dict = self._DEFAULT_CONFIGURATION
+
+        super().__init__(
+            configuration=BackendConfiguration.from_dict(config_dict), provider=provider
+        )
+
+    @classmethod
+    def _default_options(cls):
+        return Options(shots=1)
+
+    def _execute(self, data: Dict[str, Any], job_id: str = ""):
+        """Helper function to execute a job. The circuit and all relevant parameters are given in the
+        data dict. Performs validation checks on the received circuits and utilizes
+        ffsim to perform the numerical simulations.
+
+         Args:
+            data: Data dictionary that that contains the experiments to simulate, given in the shape:
+                data = {
+                    "num_species": int,
+                    "shots": int,
+                    "seed": int,
+                    "experiments": Dict[str, QuantumCircuit],
+                }
+            job_id: The job id assigned by the run method
+
+        Returns:
+            result: A qiskit job result.
+        """
+        # Start timer
+        start = time.time()
```

Comment:
Do we need this ?

Reply:
It's used later on to set the `time_taken` entry in the results dictionary. I also simply copied this from the existing [FermionSimulator](https://github.com/qiskit-community/qiskit-cold-atom/blob/0e2396f88009a370dbd25c9d029bd81859722af2/qiskit_cold_atom/fermions/fermion_simulator_backend.py#L91) backend.